torch/utils/data/dataloader.py

```diff
@@ -314,9 +314,13 @@ def _shutdown_workers(self):
             if not self.shutdown:
                 self.shutdown = True
                 self.done_event.set()
-                # if worker_manager_thread is waiting to put
-                while not self.data_queue.empty():
-                    self.data_queue.get()
+                # if worker_manager_thread is waiting to put, make place for it
+                # ignore exceptions if tensor fds are already gone
+                try:
+                    while not self.data_queue.empty():
+                        self.data_queue.get()
+                except FileNotFoundError:
```

Comment:
Can you please add a comment explaining why is getting this error ok/why can this happen?

Reply:
I have a short comment above. I'll expand it to a longer one.

Reply:
It only talks about general exceptions, while it’s nice to be more specific if you’re matching for a particular type